screens/visualization/sketch.py

```diff
@@ -165,82 +181,68 @@ def __call__(self):
-            screens = None
-        elif figure == '2':
-            # One telescope with a scattering screen.
-            tels = slice(0, 1)
-            scatters = slice(1, None)
-        elif figure in ('3', '4'):
-            # One telescope with a single scattered line
-            tels = slice(0, 1)
-            scatters = slice(1, 2)
-            if figure == '4':
-                velocity = True
-        elif figure == '5':
-            # One telescope with two opposite scatterers (no direct l.o.s.)
-            tels = slice(0, 1)
-            scatters = slice(0, 2)
-            direct = False
-            velocity = True
-        elif figure == '9':
-            tels = slice(0, 1)
-            scatters = slice(1, None)
-        else:
-            tels = slice(None)
-            scatters = slice(1, None)
-            scales = True
-
-        plt.plot(*(_v.value for _v in earth_pos), color='blue')
-        plt.plot(*(_v.value for _v in ns_pos), color='black')
-        if velocity:
-            plt.plot(*(_v.value for _v in ns_vel), color='black')
-
-        if tels:
-            plt.plot(*(_v.value for _t in tel_pos[tels] for _v in _t),
-                     color='black')
-        if direct:
-            plt.plot(*(_v.value for p2t in p2t_pos[tels] for _v in p2t),
-                     linestyle='dashed', color='black')
-        if screens:
-            plt.plot(*(_v.value for _v in screen_pos), color='grey')
-            plt.plot(*(_v.value for p2s2t in p2s2t_pos[scatters]
-                       for _p in p2s2t[tels] for _v in _p),
-                     linestyle='dotted', color='black')
-        if scales:
-            earth_bar = offset(rotate(bar(2.), 90.*u.degree), (1.3, 0.))
-            plt.plot(*(_v.value for _v in earth_bar), color='black')
-            plt.annotate(xy=(1.5, 0.), s='12000 km',
-                         verticalalignment='center')
-            off_x = ns_offset[0] - ns_size * 5
-            ns_bar = offset(rotate(bar(ns_size * 2), 90.*u.degree),
-                            (off_x, 0.))
-            plt.plot(*(_v.value for _v in ns_bar), color='black')
-            plt.annotate(xy=(off_x - 0.3, 0.), s='~20 km',
-                         verticalalignment='center',
-                         horizontalalignment='right')
-            screen_bar = offset(rotate(bar(11.8, head_length=1.5),
-                                       90*u.degree), (screen_x + 0.7, -0.5))
-            plt.plot(*(_v.value for _v in screen_bar), color='black')
-            plt.annotate(xy=(screen_x + 1., -6.), s='~10 AU',
-                         verticalalignment='center')
-            distance_bar = offset(bar(-ns_offset[0], head_length=7.),
-                                  (ns_offset[0] / 2., -5.))
-            plt.plot(*(_v for _v in distance_bar), color='black')
-            plt.annotate(xy=(ns_offset[0] / 3., -5.), s='~1 kpc',
-                         verticalalignment='center')
-
-        if filename:

```

Comment:
I think I'd keep this outside the function as well. Saving/showing is better done by the user.